stix_shifter_modules/guardium/stix_translation/query_constructor.py

```diff
@@ -2,23 +2,44 @@
 import json
 import datetime
 import copy
-from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import ObservationExpression, ComparisonExpression, \
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import ObservationExpression, \
+    ComparisonExpression, \
     ComparisonExpressionOperators, ComparisonComparators, Pattern, \
-    CombinedComparisonExpression, CombinedObservationExpression, ObservationOperators
+    CombinedComparisonExpression, CombinedObservationExpression, ObservationOperators, StartStopQualifier, SetValue
 from stix_shifter_modules.guardium.stix_translation.transformers import TimestampToGuardium
 from stix_shifter_utils.stix_translation.src.json_to_stix import observable
 from stix_shifter_utils.utils.file_helper import read_json
 from stix_shifter_utils.utils import logger
 
-
 DEFAULT_DAYS_BACK = 2
 
 
 class QueryStringPatternTranslator:
+    # Change comparator values to match with supported data source operators
+    comparator_lookup = {
+        ComparisonExpressionOperators.And: "AND",
+        ComparisonExpressionOperators.Or: "OR",
+        #        ComparisonComparators.GreaterThan: ">",
+        #        ComparisonComparators.GreaterThanOrEqual: ">=",
+        #        ComparisonComparators.LessThan: "<",
+        #        ComparisonComparators.LessThanOrEqual: "<=",
+        ComparisonComparators.Equal: "=",
+        #        ComparisonComparators.NotEqual: "!=",
+        #        ComparisonComparators.Like: "LIKE",
+        #        ComparisonComparators.In: "IN",
+        #        ComparisonComparators.Matches: 'LIKE',
+        # ComparisonComparators.IsSubSet: '',
+        # ComparisonComparators.IsSuperSet: '',
+        ObservationOperators.Or: 'OR',
+        # Treat AND's as OR's -- Unsure how two ObsExps wouldn't cancel each other out.
+        ObservationOperators.And: 'OR',
+        ComparisonComparators.In: "=",
+        # ObservationOperators.In: 'OR'
+
+    }
 
     def __init__(self, pattern: Pattern, data_model_mapper, options, transformers):
         self.dmm = data_model_mapper
-        self.comparator_lookup = self.dmm.map_comparator()
```

Comment:
This needs to remain since the comparator_lookup is now defined in the data model mapper rather than calling on the `compator_lookup` dictionary at the top of the query_constructor.

Reply:
Done the change.